main/lsp/LSPPreprocessor.cc

```diff
@@ -360,7 +360,7 @@ LSPPreprocessor::canonicalizeEdits(uint32_t v, unique_ptr<DidChangeTextDocumentP
     if (config->isUriInWorkspace(uri)) {
         string localPath = config->remoteName2Local(uri);
         if (!config->isFileIgnored(localPath)) {
-            string fileContents = changeParams->getSource(getFileContents(localPath));
+            string fileContents = changeParams->getSource(getFileContents(localPath, true));
```

Comment:
I'm not a fan of boolean parameters

what do you think about adding a method called `maybeGetFileContents` that returns an `optional<string_view>` if the file doesn't exist, and call that method instead of passing through the boolean?